pkg/action/upgrade.go

```diff
@@ -211,19 +211,11 @@ func (u *UpgradeAction) mountRWPartitions(cleanup *utils.CleanStack) error {
 	}
 	cleanup.Push(umount)
 
-	if !elemental.IsRecoveryMode(u.cfg.Config) {
-		umount, err = elemental.MountRWPartition(u.cfg.Config, u.spec.Partitions.Recovery)
-		if err != nil {
-			return elementalError.NewFromError(err, elementalError.MountRecoveryPartition)
-		}
-		cleanup.Push(umount)
-	} else {
-		umount, err = elemental.MountRWPartition(u.cfg.Config, u.spec.Partitions.State)
-		if err != nil {
-			return elementalError.NewFromError(err, elementalError.MountStatePartition)
-		}
-		cleanup.Push(umount)
+	umount, err = elemental.MountRWPartition(u.cfg.Config, u.spec.Partitions.State)
```

Comment:
This is likely to produce an undesired effect or even fail, it is likely to attempt mounting the current root as RW. By default `/run/initramfs/elemental-state` is already mounted as RW on a running system with the current setup. So it should not be needed to actually mount it as RW, it already is as RW.

Also, on a running system on btrfs, the active subvolume is mounted by default on this partition and it could also be the reported mount point from ghw library. This is the actual reason this was only called in recovery mode in the former code. Probably we need to move or add the logic at https://github.com/rancher/elemental-toolkit/blob/95389608574d77b25d583c1acbcac714ab5ee8d7/pkg/snapshotter/btrfs.go#L736-L761 into `utils.GetAllPartitions` method so we can have a known and consistent mountpoint defined for state partition.
The problem is that current Partitions struct and the underlaying ghw library assumes a single mountpoint per partition, while there could be more. In fact in the upgrade pod mountpoints under `/host/run` and `/run` are duplicated, we have not criteria from elemental-toolkit to distinguish them or even reporting them.
At mid term I'd probably drop ghw library use in favor of an `lsblk` wrapper (we used to have that at a time) and report at list of mountpoints per partition. But this is relatively ambitious refactor... specially in terms of testing.

Reply:
Interesting. I actually changed this to always remount it because the CI tests were failing because `/run/initramfs/elemental-state` could not be found. I will revert this change here, most likely this is not the solution if there was an issue with it. 
I also noticed the state partition was already RW mounted, this is why I added the debug log somewhere else to confirm this. Then the test just worked. :shrug: